JitsiConference.js

```diff
@@ -505,34 +536,52 @@ JitsiConference.prototype.replaceTrack = function (oldTrack, newTrack) {
             return Promise.reject(
                 new JitsiTrackError(JitsiTrackErrors.TRACK_IS_DISPOSED));
         }
-        // Set up the ssrcHandler for the new track before we add it at the lower levels
-        newTrack.ssrcHandler = function (conference, ssrcMap) {
-            if (ssrcMap[this.getMSID()]) {
-                this._setSSRC(ssrcMap[this.getMSID()]);
-                conference.room.removeListener(XMPPEvents.SENDRECV_STREAMS_CHANGED,
-                    this.ssrcHandler);
-            }
-        }.bind(newTrack, this);
-        this.room.addListener(XMPPEvents.SENDRECV_STREAMS_CHANGED,
-            newTrack.ssrcHandler);
     }
     // Now replace the stream at the lower levels
-    return this.room.replaceStream (oldTrack, newTrack)
+    return this._doReplaceTrack(oldTrack, newTrack)
         .then(() => {
             if (oldTrack) {
-                this.onTrackRemoved(oldTrack);
+                this.onLocalTrackRemoved(oldTrack);
             }
             if (newTrack) {
                 // Now handle the addition of the newTrack at the JitsiConference level
                 this._setupNewTrack(newTrack);
             }
             return Promise.resolve();
         }, (error) => {
+            if (newTrack) {
+                // NOTE It's is unclear from this method perspective on where is
+                // the local track added, but it is added in _setupNewTrack
+
+                // Remove the local stream if the process fails
+                this.rtc.removeLocalTrack(newTrack);
+            }
             return Promise.reject(new Error(error));
         });
 };
 
 /**
+ * Replaces the tracks at the lower level by going through the Jingle session
+ * and WebRTC peer connection. The method will resolve immediately if there is
+ * currently no JingleSession started.
+ * @param {JitsiLocalTrack|null} oldTrack the track to be removed during
+ * the process or <tt>null</t> if the method should act as "add track"
+ * @param {JitsiLocalTrack|null} newTrack the new track to be added or
+ * <tt>null</tt> if the method should act as "remove track"
+ * @return {Promise} resolved when the process is done or rejected with a string
+ * which describes the error.
+ * @private
+ */
+JitsiConference.prototype._doReplaceTrack = function (oldTrack, newTrack) {
+    if (this.jingleSession) {
+        return this.jingleSession.replaceTrack(oldTrack, newTrack);
+    } else {
+        logger.info("_doReplaceTrack - no JVB JingleSession");
```

Comment:
No, it can be only Jvb JingleSession and maybe it should be renamed this way. P2P session is stored in P2PEnabledConference